### PR TITLE
Support general attribute limits for spans

### LINKED ISSUE #2675: Support OTEL_ATTRIBUTE_VALUE_LENGTH_LIMIT and OTEL_ATTRIBUTE_COUNT_LIMIT
### Description

Currently the SDK supports the `OTEL_SPAN_ATTRIBUTE_VALUE_LENGTH_LIMIT`, but not `OTEL_ATTRIBUTE_VALUE_LENGTH_LIMIT`. Similarly `OTEL_SPAN_ATTRIBUTE_COUNT_LIMIT` is supported, but `OTEL_ATTRIBUTE_COUNT_LIMIT` is not.

Precedence needs to follow [the specification](https://github.com/open-telemetry/opentelemetry-specification/blob/d6bcc0cb072d8d6f6ced856f1f23c451648a3caa/specification/common/common.md#attribute-limits):

> An SDK MAY implement model-specific limits, for example SpanAttributeCountLimit. If both a general and a model-specific limit are implemented, then the SDK MUST first attempt to use the model-specific limit, if it isn't set, then the SDK MUST attempt to use the general limit. If neither are defined, then the SDK MUST try to use the model-specific limit default value, followed by the global limit default value.

### PR DESCRIPTION
**Pull request by MrAlias**
Fix #2675

When model specific limits are not set fallback to the general ones defined by environment variables before defaults for attribute length and count limits. This is in compliance with the specification.

#### Testing

Refactor `env` package tests to cover all environment variable helper functions, including the ones added in this PR.

### CODE REVIEW

**Review by Aneurysm9 — APPROVED** [review]

**Review by hanyuancheung — APPROVED** [review]

**Review by XSAM — APPROVED** [review]
lgtm

File: sdk/internal/env/env.go [hunk 1]
```diff
@@ -41,31 +41,61 @@ const (
...
+	// Maximum allowed span link count.
 	SpanLinkCountKey = "OTEL_SPAN_LINK_COUNT_LIMIT"
 
 	// SpanLinkAttributeCountKey
-	// Maximum allowed attribute per span link count
+	// Maximum allowed attribute per span link count.
 	SpanLinkAttributeCountKey = "OTEL_LINK_ATTRIBUTE_COUNT_LIMIT"
 )
 
+// firstInt returns the first set int value for an environment variable with
```
Comment (plantfansam) [inline sdk/internal/env/env.go:77]: Take it with a large grain of salt (this is my first review on this project!), but I might consider rephrasing to something like:

> firstInt iterates over each key and checks for an environment variables with the key name; the function returns the first one that can be cast to an int. If no integer is found (or if a non-integer is stored in a key), the function returns defaultValue.

**Review by plantfansam — APPROVED** [review]
Take-it-or-leave-it docs nit but LGTM!

File: sdk/internal/env/env.go (hunk 1 above)
Comment (MrAlias) [reply sdk/internal/env/env.go]: Updated.

**Pull request merged**